File: Data Structures & Algorithms/trapping-rain-water/submission-1.py
class Solution:
    def trap(self, height: List[int]) -> int:
        if not height:
            return 0
        stack = []
        wmax = 0
        for i in range(len(height)):
            while stack and height[i] >= height[stack[-1]]:
                mid = height[stack.pop()]
                if stack:
                    print(height[i], height[stack[-1]], mid, i, stack[-1])
                    curw = (min(height[i], height[stack[-1]])-mid) * (i - stack[-1] - 1)
                    wmax += curw
            stack.append(i)
        return wmax

File: Data Structures & Algorithms/binary-search/submission-1.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0, r = nums.size();
        while(l <= r){
            int mid = (l+r)/2;
            if(nums[mid] > target){
                cout << nums[mid] << endl;
                r = mid-1;
            } else if(nums[mid] < target){
                cout << nums[mid] << endl;
                l = mid+1;
            } else {
                cout << nums[mid] << endl;
                return mid;
            }
        }
        return -1;
    }
};
